wys/api/python/wys_google_sheet.py

```diff
@@ -83,12 +83,56 @@ def pull_from_sheet(con, service, dict_table, ward, *args):
         else:
             LOGGER.info('%s does not have any row with different date format', table_name)
 
-    insert = sql.SQL('''INSERT INTO {}.{} (ward_no, location, from_street, to_street, direction, installation_date, removal_date,
-                       new_sign_number, comments, confirmed) VALUES %s 
-                       ON CONFLICT (installation_date, new_sign_number) DO UPDATE SET 
-                       removal_date=EXCLUDED.removal_date, new_sign_number=EXCLUDED.new_sign_number, comments=EXCLUDED.comments
-                       ''').format(sql.Identifier(schema_name), sql.Identifier(table_name)) 
-    LOGGER.info('Uploaded %s rows to PostgreSQL for %s', len(rows), table_name)
+    insert = sql.SQL('''
+        WITH new_data (ward_no, location, from_street, to_street, direction, 
+                    installation_date, removal_date, new_sign_number, comments, 
+                    confirmed) 
+                    AS (
+                VALUES %s) 
+        , dupes AS(
+            /*Identify rows that will violate the unique constraint in the 
+            upsert*/
+            INSERT INTO wys.mobile_sign_installations_dupes
+            SELECT ward_no::INT, location, from_street, to_street, direction, 
+            installation_date, removal_date, new_sign_number, comments
+            FROM new_data
+            NATURAL JOIN (SELECT new_sign_number, installation_date
+                    FROM new_data
+                    GROUP BY new_sign_number, installation_date
+                    HAVING COUNT(*)> 1) dupes
```

Comment:
Beyond duplicates of `new_sign_number, installation_date` I think we could identify where `installation_date` and `removal_date` overlap for a same `new_sign_number`